vmaas/vulnerabilities.go

```diff
@@ -262,7 +258,20 @@ func repos2definitions(c *Cache, r *Request) map[DefinitionID]CpeID {
 				if r.Basearch != nil && c.RepoDetails[repoID].Basearch != *r.Basearch {
 					continue
 				}
-				if r.Releasever != nil && c.RepoDetails[repoID].Releasever != *r.Releasever {
+				candidateReleasever := c.RepoDetails[repoID].Releasever
+				if r.Releasever != nil && candidateReleasever != *r.Releasever {
+					parsedRequestReleasever, err := version.NewVersion(*r.Releasever)
+					if err != nil {
+						continue
+					}
+					parsedCandidateReleasever, err := version.NewVersion(candidateReleasever)
+					if err != nil {
+						continue
+					}
+					// Save repositories with higher releasever
+					if parsedCandidateReleasever.GreaterThan(parsedRequestReleasever) {
```

Comment:
go-version is using SemVer format and releasever does not have to be following such versioning. There is a problem with releasevers as 7Server, 7Workstation, 7ComputeNode. These should be always newer, right?
```
> x71, _ := version.NewVersion("7.1")
> x7Server, _ := version.NewVersion("7Server")
> x71.GreaterThan(x7Server)
: true
```

what about repos with no releasever? I see some repos with NULL releasever in vmaas dump and these will have empty string in `c.RepoDetails[repoID].Releasever`, parsing of empty string into version will fail
```
> empty, err := version.NewVersion("")
> err
: Malformed version:
```
Do we want to consider repos without releasever as newer or not?

Reply:
I agree that the go-version comparison is different that we'd expect in this case. But such case shouldn't happen, there is no 7Server or similar releasever for EUS/E4S/AUS streams, and this code is called only for these repositories.

The worst case scenario if there is no releasever we can't parse (it shouldn't because this code is called only for repos which have it) is that it'll behave like it's behaving now.